projects/packages/blaze/src/class-blaze.php

```diff
@@ -63,10 +63,21 @@ public function register() {
 	 * @return bool
 	 */
 	public static function should_initialize() {
-		$should_initialize = true;
+		$should_initialize  = true;
+		$jetpack_connection = new Jetpack_Connection();
 
 		// These features currently only work on WordPress.com, so they should be connected for best experience.
-		if ( ! ( new Jetpack_Connection() )->is_user_connected() ) {
+		if ( ! $jetpack_connection->is_user_connected() ) {
+			$should_initialize = false;
+		}
+
+		// We currently do not show the UI for non-English WordPress.com users.
+		$user_data = $jetpack_connection->get_connected_user_data();
```

Comment:
This method also checks for `->is_user_connected()`. It could be simplified by defining it as a var, bailing if falsey as a replacement for the check above, and remove the first `! empty( $user_data )` in this one.

Reply:
That's a good idea. I did that in 5b7524ff363a3678c554c50eed308eb427de5cd3